ECAL AlcaReco datasets definition

shervin86 opened the pull request:
Fixed runTheMatrix failures in PR #7923

Commit:
definition of EcalUncalElectron, EcalCalElectron and EcalRecalElectron sequences, dataformats and selections. remove unuseful files

Commit:
updated documentation with instructions for tests

cmsbuild commented:
A new Pull Request was created by @shervin86  for CMSSW_7_5_X.

ECAL AlcaReco datasets definition

It involves the following packages:

Calibration/EcalAlCaRecoProducers
Calibration/EcalCalibAlgos
Configuration/AlCa
Configuration/EventContent
Configuration/StandardSequences


@diguida, @cerminar, @cmsbuild, @franzoni, @nclopezo, @mmusich, @davidlange6 can you please review it and eventually sign? Thanks.
@ghellwig, @Martin-Grunewald, @makortel, @GiacomoSguazzoni, @rovere, @argiro, @appeltel, @cerati, @VinInn, @dgulhan this is something you requested to watch as well.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@nclopezo you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

diguida commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
Tested at: b1f081decfa9568e46691dcb0e05a61a9a3ea1a0
 When I ran the RelVals I found an error in the following worklfows: 
 25.0 step5
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECO+HARVEST+ALCATT/step5_TTbar+TTbar+DIGI+RECO+HARVEST+ALCATT.log

</pre>

 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8306/6685/summary.html

shervin86 commented:
Am I supposed to change also this file Configuration/PyReleaseValidation/python/relval_steps.py ?
Line 1074: 
```
                  '-s'            : 'ALCA:TkAlMuonIsolated+TkAlMinBias+EcalCalElectron+HcalCalIsoTrk+MuAlOverlaps',
```
EcalCalElectron should be removed or replaced by EcalCalZElectron+EcalCalWElectron+EcalCalZSCElectron?

shervin86 commented:
I have fixed the file Configuration/PyReleaseValidation/python/relval_steps.py
can alca double check please?

cmsbuild commented:
Pull request #8306 was updated. @diguida, @boudoul, @franzoni, @cerminar, @cmsbuild, @srimanob, @nclopezo, @mmusich, @davidlange6 can you please check and sign again.

mmusich commented:
@shervin86: looks ok to me. PdmV managers (@franzoni @srimanob @boudoul) who have signature over `Configuration/PyReleaseValidation` are also cc'ed, so they can comment

mmusich commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
Tested at: 3d3927d435e0865a335eff56ed2e834014bd034f
 I found a compilation error while trying to compile with clang: 
 
 <pre>>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-03-17-0200/src/Calibration/EcalAlCaRecoProducers/src/ValueMapTraslator.cc 
warning: unknown warning option '-Wno-psabi' [-Wunknown-warning-option]
Entering library rule at src/Calibration/EcalCalibAlgos/plugins
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-03-17-0200/src/Calibration/EcalCalibAlgos/src/ClusterFillMap.cc 
warning: unknown warning option '-Wno-psabi' [-Wunknown-warning-option]
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-03-17-0200/src/Calibration/EcalAlCaRecoProducers/plugins/EcalAlCaESAlignTrackReducer.cc:96:23: error: unused variable 'newTrackExtra' [-Werror,-Wunused-variable]
          reco::TrackExtra & newTrackExtra =  newGeneralTracksExtraCollection->back ();
                             ^
In file included from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-03-17-0200/src/Calibration/EcalAlCaRecoProducers/plugins/AlCaElectronTracksReducer.cc:1:
In file included from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-03-17-0200/src/Calibration/EcalAlCaRecoProducers/plugins/AlCaElectronTracksReducer.h:33:
In file included from /afs/cern.ch/cms/sw/ReleaseCandidates/vol0/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-03-17-0200/src/FWCore/Framework/interface/MakerMacros.h:4:
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8306/3543/summary.html

shervin86 commented:
In the last push, there were few commits that were not supposed to be there... sorry.
```
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-03-17-0200/src/Calibration/EcalAlCaRecoProducers/plugins/EcalAlCaESAlignTrackReducer.cc:96:23: error: unused variable 'newTrackExtra' [-Werror,-Wunused-variable]
```
This error should be solved now.

cmsbuild commented:
Pull request #8306 was updated. @diguida, @boudoul, @franzoni, @cerminar, @cmsbuild, @srimanob, @nclopezo, @mmusich, @davidlange6 can you please check and sign again.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8306/3543/summary.html

shervin86 commented:
there seems to be something wrong in the tests: the file EcalAlCaESAlignTrackReducer.cc should not be there anymore...

mmusich commented:
@shervin86 I believe these are the results before the changes.

mmusich commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 75208e0203b8a2c3c790296d743b4d845b21eec7
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8306/3550/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8306/3550/summary.html
There are some workflows for which there are errors in the baseline:
 401.0 step 1
 The results for the comparisons for these workflows could be incomplete 
This means most likely that the IB is having errors in the relvals

shervin86 commented:
```
----- Begin Fatal Exception 19-Mar-2015 16:56:03 CET-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing run: 1 lumi: 1 event: 1
   [1] Running path 'reconstruction_step'
   [2] Calling event method for module GoodSeedProducer/'trackerDrivenElectronSeedsTmp'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: std::vector<reco::Track>
Looking for module label: generalTracksBeforeMixing
Looking for productInstanceName: 

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound
```
I'm sorry but I don't really understand how it can be related...

shervin86 commented:
It's failing even in IB without this PR CMSSW_7_5_X_2015-03-18-0200/src , can you double check?
If it is true, can you merge this PR?

shervin86 commented:
Any update?

diguida commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 75208e0203b8a2c3c790296d743b4d845b21eec7
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8306/6854/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8306/6854/summary.html

franzoni commented:
hello @shervin86 ,

the swap in of EcalCalZElectron+EcalCalWElectron is an anticipation of  https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/1643.html . Good !

There was some discussion about this during your report at the PPD General meeting: what is the update size/event of those two producers ?

Cheers, g.

mmusich commented:
+1

shervin86 commented:
Dear Giovanni,
sorry for the late answer.

We have 2 dataformats:
  - EcalUncalElectron
  - EcalCalElectron
for each of them there are three selections:
  - W selection
  - Z selection
  - ZSC selection (one electron in tracker acceptance and the other with 2.5<|eta|<3)
for a total of 6 paths grouped in
EcalCalZElectron: Z + ZSC selection
ECalCalWElectron: W selection
EcalUncalZElectron: Z + ZSC selection
EcalUncalWElectron: W selection


Here the updated numbers:
Event size;
  EcalCalZElectron (EcalCalWElectron): 55kB/ev (40kB/ev)
  EcalUncalZElectron (EcalUncalWElectron): 95 kB/ev (80kB/ev)

Reduction factor w.r.t. PD (estimated size with 100kB/ev)
  - SingleElectron -> W selection:     24% -> 7   TB for full 2012 (using very loose ID)
  - DoubleElectron -> Z+ZSC selection: 14% -> 1.4 TB for full 2012 (using very loose ID)

I have tried to use a tigher ID for the W but it seems that running on 2012 data this will not change reduction factor significantly.
We would like to keep loose deltaEta and deltaPhi cuts to avoid inefficiencies due to mis-alignment and we have removed the E/p cut since we use the variable to calibration and we would like to avoid biases due to that.

A tightning of the selection can be considered for the 25 ns data taking to be evaluated and implemented in a further PR.
Best regards,
-- 
 							Shervin


On Thu, 2 Apr 2015, Giovanni Franzoni wrote:

> Date: Thu, 2 Apr 2015 16:10:30 +0200
> From: Giovanni Franzoni <notifications@github.com>
> Reply-To: cms-sw/cmssw
>     <reply+0045c22552d64c7bba6d16ecfaf7c988ae6bba61c368a75192cf0000000111350fd
>     692a169ce03aebaa4@reply.github.com>
> To: cms-sw/cmssw <cmssw@noreply.github.com>
> Cc: shervin86 <shervin@cern.ch>
> Subject: Re: [cmssw] ECAL AlcaReco datasets definition (#8306)
> 
> hello @shervin86 ,
>
> the swap in of EcalCalZElectron+EcalCalWElectron is an anticipation of  https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/1643.html . Good !
>
> There was some discussion about this during your report at the PPD General meeting: what is the update size/event of those two producers ?
>
> Cheers, g.
>
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/cms-sw/cmssw/pull/8306#issuecomment-88919916
>

davidlange6 commented on Calibration/EcalAlCaRecoProducers/plugins/AlCaElectronTracksReducer.cc:
```diff
@@ -0,0 +1,98 @@
+  
+  // Get GSFElectrons
+  Handle<reco::GsfElectronCollection> pElectrons;
+  iEvent.getByLabel(electronLabel_, pElectrons);
```
please complete the consumes migration for this new plugin

davidlange6 commented on Calibration/EcalAlCaRecoProducers/plugins/PUDumper.cc:
```diff
@@ -0,0 +1,74 @@
+  // get the PU collection
+  edm::Handle<std::vector<PileupSummaryInfo> > PupInfo;
+  if( !iEvent.isRealData() ){
+    iEvent.getByLabel(edm::InputTag("addPileupInfo"), PupInfo); ///\todo put it in the cfi!
```
and this one too..

shervin86 commented:
Dear David, thanks for the remarks. I've updated to one recent IB and I fixed the consumes.

cmsbuild commented:
Pull request #8306 was updated. @civanch, @diguida, @lveldere, @danduggan, @boudoul, @ssekmen, @mdhildreth, @cmsbuild, @srimanob, @Dr15Jones, @cerminar, @deguio, @franzoni, @mmusich, @davidlange6, @mulhearn, @nclopezo can you please check and sign again.

cmsbuild commented:
+1
Tested at: e9548823af7e9bce8063924e322573630f3b7b0a
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8306/4491/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8306/4491/summary.html

deguio commented:
+1
not related to DQM

lveldere commented:
will you rebase?

shervin86 commented:
If I rebase, do I need to do another PR? And then should I wait again for tests and approvals?
How long should this take?

-- 
 							Shervin


On Tue, 5 May 2015, lveldere wrote:

> Date: Tue, 5 May 2015 11:11:12 +0200
> From: lveldere <notifications@github.com>
> Reply-To: cms-sw/cmssw
>     <reply+0045c2257a82b1add985d04990d547a4b3b2f0d3d57d656792cf0000000111604b3
>     092a169ce03aebaa4@reply.github.com>
> To: cms-sw/cmssw <cmssw@noreply.github.com>
> Cc: shervin86 <shervin@cern.ch>
> Subject: Re: [cmssw] ECAL AlcaReco datasets definition (#8306)
> 
> will you rebase?
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/cms-sw/cmssw/pull/8306#issuecomment-99003331
>

lveldere commented:
no, if you rebase and then push,
your pr will be updated automatically
Lukas

On Tue, May 5, 2015 at 11:18 AM, shervin86 <notifications@github.com> wrote:

> If I rebase, do I need to do another PR? And then should I wait again for
> tests and approvals?
> How long should this take?
>
> --
> Shervin
>
>
> On Tue, 5 May 2015, lveldere wrote:
>
> > Date: Tue, 5 May 2015 11:11:12 +0200
> > From: lveldere <notifications@github.com>
> > Reply-To: cms-sw/cmssw
> >
> <reply+0045c2257a82b1add985d04990d547a4b3b2f0d3d57d656792cf0000000111604b3
> > 092a169ce03aebaa4@reply.github.com>
> > To: cms-sw/cmssw <cmssw@noreply.github.com>
> > Cc: shervin86 <shervin@cern.ch>
> > Subject: Re: [cmssw] ECAL AlcaReco datasets definition (#8306)
> >
> > will you rebase?
> >
> > ---
> > Reply to this email directly or view it on GitHub:
> > https://github.com/cms-sw/cmssw/pull/8306#issuecomment-99003331
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/8306#issuecomment-99006159>.
>

shervin86 commented:
Ok, since it is not possible to push the rebased branch since it is not a fast-forward, I will remove the branch and push the rebased one with the same name.
Should this work?

-- 
 							Shervin


On Tue, 5 May 2015, lveldere wrote:

> Date: Tue, 5 May 2015 11:21:49 +0200
> From: lveldere <notifications@github.com>
> Reply-To: cms-sw/cmssw
>     <reply+0045c2252d06dcfff74e5623cf9c6aa3071581a85dd7597a92cf0000000111604da
>     d92a169ce03aebaa4@reply.github.com>
> To: cms-sw/cmssw <cmssw@noreply.github.com>
> Cc: shervin86 <shervin@cern.ch>
> Subject: Re: [cmssw] ECAL AlcaReco datasets definition (#8306)
> 
> no, if you rebase and then push,
> your pr will be updated automatically
> Lukas
>
> On Tue, May 5, 2015 at 11:18 AM, shervin86 <notifications@github.com> wrote:
>
>> If I rebase, do I need to do another PR? And then should I wait again for
>> tests and approvals?
>> How long should this take?
>>
>> --
>> Shervin
>>
>>
>> On Tue, 5 May 2015, lveldere wrote:
>>
>>> Date: Tue, 5 May 2015 11:11:12 +0200
>>> From: lveldere <notifications@github.com>
>>> Reply-To: cms-sw/cmssw
>>>
>> <reply+0045c2257a82b1add985d04990d547a4b3b2f0d3d57d656792cf0000000111604b3
>>> 092a169ce03aebaa4@reply.github.com>
>>> To: cms-sw/cmssw <cmssw@noreply.github.com>
>>> Cc: shervin86 <shervin@cern.ch>
>>> Subject: Re: [cmssw] ECAL AlcaReco datasets definition (#8306)
>>>
>>> will you rebase?
>>>
>>> ---
>>> Reply to this email directly or view it on GitHub:
>>> https://github.com/cms-sw/cmssw/pull/8306#issuecomment-99003331
>>
>>>
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/cms-sw/cmssw/pull/8306#issuecomment-99006159>.
>>
>
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/cms-sw/cmssw/pull/8306#issuecomment-99006529

lveldere commented:
git push -f HEAD:<YOUR_BRANC_NAME>

the -f does the trick for me usually

On Tue, May 5, 2015 at 11:50 AM, shervin86 <notifications@github.com> wrote:

> Ok, since it is not possible to push the rebased branch since it is not a
> fast-forward, I will remove the branch and push the rebased one with the
> same name.
> Should this work?
>
> --
> Shervin
>
>
> On Tue, 5 May 2015, lveldere wrote:
>
> > Date: Tue, 5 May 2015 11:21:49 +0200
> > From: lveldere <notifications@github.com>
> > Reply-To: cms-sw/cmssw
> >
> <reply+0045c2252d06dcfff74e5623cf9c6aa3071581a85dd7597a92cf0000000111604da
>
> > d92a169ce03aebaa4@reply.github.com>
> > To: cms-sw/cmssw <cmssw@noreply.github.com>
> > Cc: shervin86 <shervin@cern.ch>
> > Subject: Re: [cmssw] ECAL AlcaReco datasets definition (#8306)
> >
> > no, if you rebase and then push,
> > your pr will be updated automatically
> > Lukas
> >
> > On Tue, May 5, 2015 at 11:18 AM, shervin86 <notifications@github.com>
> wrote:
> >
> >> If I rebase, do I need to do another PR? And then should I wait again
> for
> >> tests and approvals?
> >> How long should this take?
> >>
> >> --
> >> Shervin
> >>
> >>
> >> On Tue, 5 May 2015, lveldere wrote:
> >>
> >>> Date: Tue, 5 May 2015 11:11:12 +0200
> >>> From: lveldere <notifications@github.com>
> >>> Reply-To: cms-sw/cmssw
> >>>
> >>
> <reply+0045c2257a82b1add985d04990d547a4b3b2f0d3d57d656792cf0000000111604b3
> >>> 092a169ce03aebaa4@reply.github.com>
> >>> To: cms-sw/cmssw <cmssw@noreply.github.com>
> >>> Cc: shervin86 <shervin@cern.ch>
> >>> Subject: Re: [cmssw] ECAL AlcaReco datasets definition (#8306)
> >>>
> >>> will you rebase?
> >>>
> >>> ---
> >>> Reply to this email directly or view it on GitHub:
> >>> https://github.com/cms-sw/cmssw/pull/8306#issuecomment-99003331
> >>
> >>>
> >>
> >> —
> >> Reply to this email directly or view it on GitHub
> >> <https://github.com/cms-sw/cmssw/pull/8306#issuecomment-99006159>.
> >>
> >
> >
> > ---
> > Reply to this email directly or view it on GitHub:
> > https://github.com/cms-sw/cmssw/pull/8306#issuecomment-99006529
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/8306#issuecomment-99014371>.
>

cmsbuild commented:
Pull request #8306 was updated. @diguida, @boudoul, @franzoni, @cerminar, @cmsbuild, @srimanob, @nclopezo, @mmusich, @davidlange6 can you please check and sign again.

mmusich commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: dc78a86608f9f45a27b350152a857f3474c3a608
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8306/14/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/c8e06956a48d7cbbd617258091d03287290d2da7
https://github.com/cms-sw/cmssw/commit/6ca740aa79f9a5be96404772f239791a5591f670
https://github.com/cms-sw/cmssw/commit/2f51210b07f31c1a14d649a137e049c644c90701
https://github.com/cms-sw/cmssw/commit/5b137e38a83b79d557a93f325b0fb26ddc5091ba
https://github.com/cms-sw/cmssw/commit/fff27f36a499fac894a87ce8e17e66c38899bd57
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8306/14/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8306/14/git-merge-result

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8306/14/summary.html

The workflows 140.53 have different files in step1_dasquery.log than the ones found in the baseline. You may want to check and retrigger the tests if necessary. You can check it in the "files" directory in the results of the comparisons

mmusich commented:
@shervin86 now this doesn't merge anymore...

shervin86 commented:
It's not my fault. Somebody changed AlCaRecoStreams_cff in the meanwhile.
I can push a new rebase, it would be preferable to merge it very soon

cmsbuild commented:
Pull request #8306 was updated. @diguida, @boudoul, @franzoni, @cerminar, @cmsbuild, @srimanob, @nclopezo, @mmusich, @davidlange6 can you please check and sign again.

mmusich commented:
@cmsbuild please test

mmusich commented:
@shervin86 we will put on hold conflicting commits changing the AlCaRecoStreams_cff

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 9cb811f057408288bb864c775587d9bcdaa1f5a3
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8306/44/summary.html

mmusich commented:
+1

shervin86 commented:
ping orp and pdmv

emanueledimarco commented:
Can we approve these changes asap? This would help in case in the meantime changes in the package will be pushed, that would need further work on our side to adjust to the changes.

franzoni commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_5_X IBs unless changes (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @nclopezo, @smuzaffar

shervin86 commented:
when the ORP meeting is supposed to be?

mmusich commented:
@shervin86 it was supposed to happen yesterday (but skipped because of the concurring Offline/Computing workshop)
@davidlange6 can we please have it merged? Thanks

davidlange6 commented:
Hi Marco

Raise the issue on the ORP email in lieu of a meeting (as requested already)

Cheers-
David

> On May 13, 2015, at 10:17 AM, Marco Musich <notifications@github.com> wrote:
> 
> @shervin86 it was supposed to happen yesterday (but skipped because of the concurring Offline/Computing workshop)
> @davidlange6 can we please have it merged? Thanks
> 
> —
> Reply to this email directly or view it on GitHub.
>

davidlange6 commented:
+1
